core/scripts/bareos-config-lib.sh.in

```diff
@@ -997,16 +997,23 @@ apply_dbconfig_settings()
 get_local_hostname()
 {
     # put actual short hostname in configuration files
-    # try to get short hostname
-    hname=$(hostname -s)
-    if [ -z "${hname}" ]; then
-        # try to get long hostname
-        hname=$(hostname|sed 's/\..*//g')
+    if [ -z "$(command -v hostname)" ]; then
+        hname=$(hostname -s)
         if [ -z "${hname}" ]; then
-            # set to "localhost"
-            hname='localhost'
+            # try to get long hostname
+            hname=$(hostname|sed 's/\..*//g')
         fi
+    elif [ -z "$(command -v hostnamectl)" ]; then
+        hname=$(hostnamectl hostname)
```

Comment:
According to POSIX (IEEE Std 1003.1-2001) calling `uname -n` will "Write the name of this node within an implementation-defined communications network."

That would probably be more portable than calling some systemd-specific stuff. However, you'll probably still need sed to shorten the name.
Maybe we can even rely on `uname` *instead* of calling `hostname` in the first place.

Reply:
Yes excellent why we didn't thing about that first. I will propose to move uname -n and if faile cat /etc/hostname if exist and not empty, otherwise localhost.

Reply:
As `uname` is a POSIX requirement and on Linux is a part of coreutils, I guess it is fine  to just fail hard in case it is not available.

Reply:
done.